### ISSUE TITLE
softmax loss and focal loss

### ISSUE THREAD

**Issue by maryhh** [deepinsight/insightface#562]
我把这行代码` softmax = mx.symbol.SoftmaxOutput(data=fc7, label = gt_label, name='softmax', normalization='valid')`替换成`focalloss = mx.symbol.Custom(data=fc7, op_type='FocalLoss', labels = gt_label, name='focalloss', alpha=0.25, gamma=2)` 然后换了损伤函数,可是它提示在backword 的时候label越界了,为了方便测试我随便设置了batch_size = 16,num_classes = 717, 然后我就输出了一下label和pred_label,
发现label超过717了,但是我用softmaxloss output的时候label也超过717了,但是没有问题,正常运行,请问是怎么回事啊?
![default](https://user-images.githubusercontent.com/26423386/53401409-b565f800-39ea-11e9-8679-67547451fc8a.png)
![default](https://user-images.githubusercontent.com/26423386/53401562-0118a180-39eb-11e9-8a98-0b5bf9ba57be.png)

**Comment by xzxedu** [comment]
Hi, may I ask did you try to replace the OHEM with focal loss and get a better results?
